GLTFSDK.Test/Source/GLTFExtensionsTests.cpp

```diff
@@ -373,17 +440,57 @@ namespace Microsoft
                         Assert::IsTrue(textureTransform.offset == offset);
                         Assert::IsTrue(textureTransform.rotation == rotation);
                         Assert::IsTrue(textureTransform.scale == scale);
-                        Assert::IsTrue(textureTransform.texCoord == texCoord);
+                        Assert::IsTrue(
+                            (!textureTransform.texCoord && !texCoord) ||
+                            ((textureTransform.texCoord && texCoord) &&
+                                *textureTransform.texCoord == *texCoord));
                     };
 
                     Assert::IsTrue(doc.materials.Size() == 9);
 
-                    checkTextureInfo(doc.materials[0], Vector2(0.5f, 0.0f), 0.0f, Vector2(1.0f, 1.0f), 0);
-                    checkTextureInfo(doc.materials[1], Vector2(0.0f, 0.5f), 0.0f, Vector2(1.0f, 1.0f), 0);
-                    checkTextureInfo(doc.materials[2], Vector2(0.5f, 0.5f), 0.0f, Vector2(1.0f, 1.0f), 0);
-                    checkTextureInfo(doc.materials[3], Vector2(0.0f, 0.0f), 0.39269908169872415480783042290994f, Vector2(1.0f, 1.0f), 0);
-                    checkTextureInfo(doc.materials[4], Vector2(0.0f, 0.0f), 0.0f, Vector2(1.5f, 1.5f), 0);
-                    checkTextureInfo(doc.materials[5], Vector2(-0.2f, -0.1f), 0.3f, Vector2(1.5f, 1.5f), 0);
+                    checkTextureInfo(doc.materials[0], Vector2(0.5f, 0.0f), 0.0f, Vector2(1.0f, 1.0f), {}); // Note: texCoord not specified
+                    checkTextureInfo(doc.materials[1], Vector2(0.0f, 0.5f), 0.0f, Vector2(1.0f, 1.0f), {});
+                    checkTextureInfo(doc.materials[2], Vector2(0.5f, 0.5f), 0.0f, Vector2(1.0f, 1.0f), {});
+                    checkTextureInfo(doc.materials[3], Vector2(0.0f, 0.0f), 0.39269908169872415480783042290994f, Vector2(1.0f, 1.0f), {});
+                    checkTextureInfo(doc.materials[4], Vector2(0.0f, 0.0f), 0.0f, Vector2(1.5f, 1.5f), {});
+                    checkTextureInfo(doc.materials[5], Vector2(-0.2f, -0.1f), 0.3f, Vector2(1.5f, 1.5f), {});
+                }
+
+                GLTFSDK_TEST_METHOD(ExtensionsTests, Extensions_Test_HasTextureTransformExtension_TexCoord)
+                {
+                    // Ensure the optionality of the texCoord property is preserved
+                    const auto extensionDeserializer = KHR::GetKHRExtensionDeserializer();
+                    auto doc = Deserialize(extensionSchemaKHRTextureTransform_TexCoord, extensionDeserializer);
+
+                    auto checkTextureInfo = [](
```

Comment:
You should pass in a TextureTransform object and use its equality operator instead of copying this code everywhere